staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -456,6 +456,15 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+// TODO: currently strict validation is set via
+// query param. Alternatively we could use a request header.
+// TODO: maybe we should check content-type here and error for protobuf (ie take the full req instead of just the URL?)
```

Comment:
If it doesn't work, it needs to return an error.

Reply:
First pass at doing this.